src/bootstrap/compile.rs

```diff
@@ -1063,8 +1088,12 @@ pub fn run_cargo(builder: &Builder,
             let filename = Path::new(&*filename);
 
             // If this was an output file in the "host dir" we don't actually
-            // worry about it, it's not relevant for us.
+            // worry about it, it's not relevant for us
             if filename.starts_with(&host_root_dir) {
+                // Unless it's a proc macro used in the compiler
+                if package_id.starts_with("rustc_macros ") {
```

Comment:
To double-check this, should this be a whitelist which also includes `serde_derive`?

(I'm sort of hoping that CI fails because othrewise this wouldn't match my expectations, but let's leave CI running and see what happens)

Reply:
I missed that. It won't get far without the right crate here ;)

Reply:
Phew, ok good to know the world still makes a little sense!